### PR TITLE
Show image for patients that are in Darwin

### PR DESCRIPTION
**Pull request by inodb**
screen shot 2017-03-22 at 5 17 12 pm

This `darwinAccessUrl` parameter is currently injected in the `patient_view.jsp` from `cbioportal/cbioportal`. Therefore we can expect it to be there. Eventually we would like this to be provided through an API endpoint.

darwin

### CODE REVIEW

**Comment by inodb** [comment]
[(5) Link to Darwin in patient view (Needed by MSKCC)](https://trello.com/c/4OH73UxL/95-5-link-to-darwin-in-patient-view-needed-by-mskcc)

File: src/pages/patientView/patientHeader/PatientHeader.tsx [hunk 1]
```diff
@@ -18,11 +17,26 @@ export default class PatientHeader extends React.Component<IPatientHeaderProps,
...
+        // use JSP injected Darwin URL window.darwinAccessUrl
+        // TODO: use internal API service instead, once this exists
+        let darwinAccessUrl: any = ((window as any).darwinAccessUrl);
+
+        if (darwinAccessUrl !== undefined && darwinAccessUrl !== null && darwinAccessUrl !== '') {
+            // add link to darwin
+            let darwinImgSrc = require("./images/darwin_logo.png");
+            return (<a target={'_blank'} href={darwinAccessUrl}><img style={{paddingLeft:'5px'}} src={darwinImgSrc} /></a>);
+        } else {
+            return '';
```
Comment (alisman) [inline src/pages/patientView/patientHeader/PatientHeader.tsx]: I think return null would be better than empty string

File: src/pages/patientView/patientHeader/PatientHeader.tsx [hunk 2]
```diff
@@ -18,11 +17,26 @@ export default class PatientHeader extends React.Component<IPatientHeaderProps,
...
 
+    private getDarwinAccessUrl() {
+        // use JSP injected Darwin URL window.darwinAccessUrl
+        // TODO: use internal API service instead, once this exists
+        let darwinAccessUrl: any = ((window as any).darwinAccessUrl);
+
+        if (darwinAccessUrl !== undefined && darwinAccessUrl !== null && darwinAccessUrl !== '') {
+            // add link to darwin
+            let darwinImgSrc = require("./images/darwin_logo.png");
+            return (<a target={'_blank'} href={darwinAccessUrl}><img style={{paddingLeft:'5px'}} src={darwinImgSrc} /></a>);
```
Comment (alisman) [inline src/pages/patientView/patientHeader/PatientHeader.tsx]: you can just use "_blank" if value is not dynamic.

File: src/pages/patientView/patientHeader/PatientHeader.tsx [hunk 3]
```diff
@@ -18,11 +17,26 @@ export default class PatientHeader extends React.Component<IPatientHeaderProps,
...
+                {this.getDarwinAccessUrl()}
             </div>
         );
 
     }
 
+    private getDarwinAccessUrl() {
+        // use JSP injected Darwin URL window.darwinAccessUrl
+        // TODO: use internal API service instead, once this exists
+        let darwinAccessUrl: any = ((window as any).darwinAccessUrl);
```
Comment (alisman) [inline src/pages/patientView/patientHeader/PatientHeader.tsx]: I would do :string | null | undefined

typescript would probably be fine with just string.

**Pull request merged**